### ISSUE TITLE
Feature Proposal for IntelliJ

### ISSUE THREAD

**Issue by pakoito** [arrow-kt/arrow-meta#31]
This is a proposal issue for Ide features we might want to consider for Arrow.

Comment below for suggestions and Feedback.

**Comment by i-walker** [comment]
@ahinchman1 there is already an IR Generator within the functionality of the `Decompile to Kotlin Bytecode` Action. 

Please, correct me if that is not what you were proposing.
<img width="1190" alt="Screenshot 2019-11-21 at 12 58 23" src="https://user-images.githubusercontent.com/46971368/69336363-a0a74f80-0c5e-11ea-9034-fca26fee83c0.png">

**Comment by ahinchman1** [comment]
@i-walker yes, I found out mine is just not working hahah - I'm sure I can figure out what's wrong with it eventually

**Comment by ahinchman1** [comment]
Some validation received after giving an internal talk on AST parsing:

It would be amazing to write an IDE plugin that displays an actual tree for AST parsing unlike the PSIViewer. It's better than nothing but one day it would be amazing to see trees that look closer to this:

![Screen Shot 2019-11-22 at 1 18 25 PM](https://user-images.githubusercontent.com/6917498/69425112-96528780-0d2a-11ea-9ce6-ed63c4504a44.png)
